api/METRICS-DASHBOARD.md

```diff
@@ -0,0 +1,85 @@
+# Proposal for metrics/dashboard API
+
+Goals:
+* Unified treatment of many different kinds of metrics in dashboard
+* A clean separation between dashboard (for visualization) and different kinds of backends (for metric calculation)
+
+Status quo:
+* Dashboard currently keeps track of the metric values in the cache dictionary `fairlearn.widget._fairlearn_widget.FairlearnWidget._response`
+* The dictionary is updated in `fairlearn.widget._fairlearn_dashboard.FairlearnDashboard._on_request`
+* There's a PR out to fill the whole data structure in  `fairlearn.metrics.create_dashboard_dictionary`
+
+Issues with the status quo:
+* Code duplication / redundancy / brittleness due to copy-paste errors
+* Currently only one kind of a metric (`<metric>_summary`) is supported and hard-wired into the dashboard dictionary
+
+## Proposal
+
+### Part I: More general dashboard dictionary
+
+```python
+{ 
+  "prediction_type":  "binary_classification" or "probabilistic_binary_classification" or "regression",
+  "array_bindings": {   # all 1D arrays, including features and predictior vectors, are here
+    "<array_key>" : {   # the keys can be arbitrary strings; not sure we need to force any convention, but see examples below
+      "name": string,   # the name of a feature would be the feature name, of a prediction vector would be the model name
+      "values": number[],  
+      "value_names": string[],       # an optional field to encode categorical data   

```

Comment:
Presumably we also specify that extra keys (e.g. inserted by AzureML) are to be preserved.